src/node_url.cc

```diff
@@ -2348,6 +2348,21 @@ std::string URL::ToFilePath() const {
 #endif
 }
 
+URL URL::FromFilePath(const std::string& file_path) {
+  URL url;
+  url.context_.scheme = "file:";
```

Comment:
I'm a bit reluctant to touch into the internals in this way. Could something like this work?

```c++
URL url("file:///");
```

This will also need a cctest (see https://github.com/nodejs/node/blob/master/test/cctest/test_url.cc).

Finally could you split the changes to node\_url into a separate commit?